cogs/displayinfo.py

```diff
@@ -33,85 +32,139 @@ class DisplayInfo(commands.Cog):
+
+                if self.timed_task == None: #if task is finished create new task
+                    self.timed_task = asyncio.create_task(self.timed_update())
+                print("added:", mem_after.name)
+            
+            if self.members_online[mem_after.id] == mem_after.guild.id:
+                update_success = await self._update_now(mem_after)
+                print("start:", self.timers)
+        elif (mem_before.activity and not mem_after.activity) or (mem_before.status != mem_after.status and mem_after.status == discord.Status.offline): # activity stopped or member goes offline
+            if mem_before.id in self.members_online:
+                self.members_online.pop(mem_before.id)
+                print("removed:", mem_before.name)
+                if len(self.members_online) <= 0:
+                    self.timed_task.cancel()    
+                    self.timed_task = None
+                    
+                before_string_id = str(mem_before.id)
+                if before_string_id in self.timers:
+                    if self.current_date in self.timers[before_string_id]:
+                        if mem_before.activity and mem_before.activity.name in self.timers[before_string_id][self.current_date]:
+                            currentStartTime = self.timers[before_string_id][self.current_date][mem_before.activity.name]["startTime"]
+                            self.timers[before_string_id][self.current_date][mem_before.activity.name]["startTime"] = None
+                            if currentStartTime:
+                                playTime = datetime.now() - currentStartTime # Calculate the time difference (playTime) between the start and end time of the activity
+                                self.timers[before_string_id][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[before_string_id][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
+                            print("end:", self.timers) # printing the dictionary to see the updates after an activity is ended
+                            self.current_collection.replace_one({}, self.timers) # Update the MongoDB collection with the updated activity data(self.timers)
+
+    #This function periodically updates the activities of members that are online and in an activity.
+    async def timed_update(self):
+        while len(self.members_online) > 0:
             await asyncio.sleep(300)
+            for memberID, guildID in self.members_online.items():
+                guild = self.bot.get_guild(guildID)
+                member = guild.get_member(memberID)
+                update_success = await self._update_now(member)
+            
+            print("timed update:", self.timers)
+            self.current_collection.replace_one({}, self.timers)
+
+    #This function resets all the startTimes of a member
```

Comment:
NIT: Space after the `#`

Also we need to be careful that we say "all startTimes" when we haven't yet removed the old start times. We are currently only looking through the current date. For now, just change up the wording of the comment to mean that it will reset all the startTimes of a member on the current date.